### PR TITLE
[SPARK-39174][SQL] Catalogs loading swallows missing classname for ClassNotFoundException

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

this PR captures the actual missing classname when catalog loading meets ClassNotFoundException

### Why are the changes needed?

ClassNotFoundException can occur when missing dependencies, we shall not always report the catalog class is missing

### Does this PR introduce _any_ user-facing change?

yes, when loading catalogs and ClassNotFoundException occurs, it shows the correct missing class.

### How was this patch tested?

new test added

### CODE REVIEW

**Review by LuciferYang — APPROVED** [review]
LGTM

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala [hunk 1]
```diff
@@ -1544,8 +1544,14 @@ object QueryExecutionErrors extends QueryErrorsBase {
 
   def catalogPluginClassNotFoundForCatalogError(
       name: String,
-      pluginClassName: String): Throwable = {
-    new SparkException(s"Cannot find catalog plugin class for catalog '$name': $pluginClassName")
```
Comment (HyukjinKwon) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala:1548]: Hm, what about we just add cause to `new SparkException(..., cause = e)`?

Comment (yaooqinn) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala:1548]: SGTM

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master, branch-3.3, branch-3.2 and branch-3.1.

**Pull request closed without merging**